app/Services/Contact/Relationship/CreateRelationship.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace App\Services\Contact\Relationship;
+
+use App\Services\BaseService;
+use App\Models\Contact\Contact;
+use App\Models\Relationship\Relationship;
+use App\Models\Relationship\RelationshipType;
+
+class CreateRelationship extends BaseService
+{
+    /**
+     * Get the validation rules that apply to the service.
+     *
+     * @return array
+     */
+    public function rules()
+    {
+        return [
+            'account_id' => 'required|integer|exists:accounts,id',
+            'contact_is' => 'required|integer|exists:contacts,id',
+            'of_contact' => 'required|integer|exists:contacts,id',
+            'relationship_type_id' => 'required|integer|exists:relationship_types,id',
+        ];
+    }
+
+    /**
+     * Set a relationship between two contacts.
+     *
+     * @param array $data
+     *
+     * @return Relationship
+     */
+    public function execute(array $data) : Relationship
+    {
+        $this->validate($data);
+
+        $contact = Contact::where('account_id', $data['account_id'])
+            ->findOrFail($data['contact_is']);
+
+        $otherContact = Contact::where('account_id', $data['account_id'])
+            ->findOrFail($data['of_contact']);
+
+        $relationshipType = RelationshipType::where('account_id', $data['account_id'])
+            ->findOrFail($data['relationship_type_id']);
+
+        $reverseRelationshipType = $relationshipType->reverseRelationshipType();
```

Comment:
1. Why do you do this here and not inside the `setRelationship` method below? It's weird to read it that way, as if you are setting the reverse relationship before the normal relationship.

2. Also, why do you do an `if` here? Aren't relationships always bi-directional?

Reply:
1. Ok I have change the order. The setRelationship create one relationship only.
2. Because the tests does not always create the reverse relationshiptype object.